packages/multi-column/src/extensions/ColumnResize/ColumnResizeExtension.ts

```diff
@@ -0,0 +1,376 @@
+    // We need to scale the width change by the left column's width in
+    // percent, otherwise the rate at which the resizing happens will change
+    // based on the width of the left column.
+    const scaledWidthChangePx =
+      widthChangePx * pluginState.leftColumn.widthPercent;
+    const widthChangePercent =
+      (pluginState.leftColumn.widthPx + scaledWidthChangePx) /
+        pluginState.leftColumn.widthPx -
+      1;
+
+    let newLeftColumnWidth =
+      pluginState.leftColumn.widthPercent + widthChangePercent;
+    let newRightColumnWidth =
+      pluginState.rightColumn.widthPercent - widthChangePercent;
+
+    // Ensures that the column widths do not go below the minimum width.
+    // There is no maximum width, the user can resize the columns as much as
+    // they want provided the others don't go below the minimum width.
+    if (newLeftColumnWidth < this.COLUMN_MIN_WIDTH_PERCENT) {
+      newRightColumnWidth -= this.COLUMN_MIN_WIDTH_PERCENT - newLeftColumnWidth;
+      newLeftColumnWidth = this.COLUMN_MIN_WIDTH_PERCENT;
+    } else if (newRightColumnWidth < this.COLUMN_MIN_WIDTH_PERCENT) {
+      newLeftColumnWidth -= this.COLUMN_MIN_WIDTH_PERCENT - newRightColumnWidth;
+      newRightColumnWidth = this.COLUMN_MIN_WIDTH_PERCENT;
+    }
+
+    console.log("newLeftColumnWidth", newLeftColumnWidth);
+    console.log("newRightColumnWidth", newRightColumnWidth);
+    this.view.dispatch(
+      this.view.state.tr
+        .setNodeAttribute(
+          pluginState.leftColumn.pos,
+          "width",
+          newLeftColumnWidth
+        )
+        .setNodeAttribute(
+          pluginState.rightColumn.pos,
+          "width",
+          newRightColumnWidth
+        )
+    );
+  };
+
+  // If the plugin is in a resize state, we want to revert it to a default
+  // or hover, depending on where the mouse cursor is, when the user
+  // releases the mouse button.
+  mouseUpHandler = (event: MouseEvent) => {
+    const pluginState = columnResizePluginKey.getState(this.view.state);
+    if (!pluginState || pluginState.type !== "resize") {
+      return;
+    }
+
+    const newState = this.getColumnHoverOrDefaultState(event);
+
+    // Revert plugin state to default or hover, depending on where the mouse
+    // cursor is.
+    console.log(newState);
+    this.view.dispatch(
+      this.view.state.tr.setMeta(columnResizePluginKey, newState)
+    );
+  };
+
+  // In a `columnList` node, we expect that the average width of each column
+  // is 1. However, there are cases in which this stops being true. For
+  // example, making one really wide column and then removing it will cause
+  // the width average to go down. This isn't really an issue until the user
+  // tries to add a new column, which will, in this case, be wider than
+  // expected. Therefore, this function traverses the document and
+  // normalizes the average column width to 1 whenever and wherever
+  // necessary.
+  update = (view: EditorView) => {
```

Comment:
this is relatively heavy as it runs on every update and traverses all nodes. Could we also fix this when inserting a column for example?

Reply:
Ideally we would do it whenever removing a column (since this is when the average width across columns may change), but doing it on insert should give the same UX. The only small difference is when the document is saved the widths won't not be normalized. But again this doesn't change anything for the user and it's much less complex to do it on insert instead.

Reply:
ok, can you move this logic to when we "drop-to-new-column"?